### PR TITLE
PYTHON-3120 Set up flake8 linting

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: gridfs/grid_file.py [hunk 1]
```diff
@@ -518,7 +518,7 @@ def readchunk(self) -> bytes:
                     self, self.__chunks, self._session, chunk_number
                 )
 
-            chunk = self.__chunk_iter.next()
+            chunk = next(self.__chunk_iter)
```
[the remark this answers is not in the document]
Comment (blink1073) [reply gridfs/grid_file.py:521]: Done

File: pymongo/compression_support.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def validate_compressors(dummy, value):
 def validate_zlib_compression_level(option, value):
     try:
         level = int(value)
-    except:
+    except BaseException:
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pymongo/compression_support.py:85]: Done

File: pymongo/monitoring.py [hunk 3]
```diff
@@ -633,7 +633,7 @@ def __init__(
         super(CommandStartedEvent, self).__init__(
             command_name, request_id, connection_id, operation_id, service_id=service_id
         )
-        cmd_name, cmd_doc = command_name.lower(), command[command_name]
+        cmd_name, _ = command_name.lower(), command[command_name]
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pymongo/monitoring.py:636]: Done

File: test/mockupdb/test_slave_okay_single.py [hunk 4]
```diff
@@ -48,13 +48,11 @@ def test(self):
...
-            slave_ok = mode != "primary" or server_type != "mongos"
-        elif operation.op_type == "must-use-primary":
-            slave_ok = server_type != "mongos"
-        else:
+        if operation.op_type not in (
+            "always-use-secondary",
+            "may-use-secondary",
+            "must-use-primary",
+        ):
             assert False, "unrecognized op_type %r" % operation.op_type
```
[the remark this answers is not in the document]
Comment (blink1073) [reply test/mockupdb/test_slave_okay_single.py:56]: Done

**Comment by blink1073** [comment]
Wow, that was a head-scratcher.  `flake` had run fine locally but was failing in CI.  There were changes in `master` that made an import unused when merged into `master` in the CI run.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**